src/lib/util.ts

```diff
@@ -108,3 +114,75 @@ export function camelCaseKeys(obj: any): any {
     })
   );
 }
+
+// Helper function to enable waiting until a transaction has been materialized by the Validator.
+// Uses simple polling with exponential backoff up to a maximum timeout.
+// Potential optimization could be had if the Validator supports subscribing to transaction
+// receipts via Websockets or long-poling in the future
+export async function onConfirm(
+  this: Connection,
+  txnHash: string,
+  options?: ConfirmOptions
+): Promise<ReceiptResult> {
+  // default timeout 2 minutes
+  // TODO: move default to a single spot
+  const timeout = options?.timeout ?? 120 * 1000;
+
+  // determines how often to check for materialization before timeout
+  const rate = options?.rate ?? 1500;
+  const start = Date.now();
+
+  // next tick then try immediately
+  await new Promise((resolve) => setTimeout(resolve, 0));
+  let table = await this.receipt(txnHash);
+
+  let tries = 0;
+  while (!table && start + timeout > Date.now()) {
+    // increase the time between each call, but never go past the specified timeout
+    const waitForMs = rate * Math.pow(2, tries);
+    const nextTry =
+      start + timeout < Date.now() + waitForMs
+        ? start + timeout - Date.now()
+        : waitForMs;
+
+    await new Promise((resolve) => setTimeout(resolve, nextTry));
+    table = await this.receipt(txnHash);
+    tries++;
+  }
+
+  // Throw and let the caller decide what to do if the timeout is exceeded
+  if (!table) {
+    throw new Error(
+      `timeout exceeded: could not get transaction receipt: ${txnHash}`
+    );
+  }
+
+  return table;
+}
+
+// TODO: this could be removed if we bump major version
```

Comment:
I think we're going to have to anyway, right?

Reply:
Yep, this comment was before the combination of PRs I'll remove it